src/util/system.cpp

```diff
@@ -492,7 +492,7 @@ std::vector<std::string> ArgsManager::GetArgs(const std::string& strArg) const
 {
     std::vector<std::string> result;
     for (const util::SettingsValue& value : GetSettingsList(strArg)) {
-        result.push_back(value.isFalse() ? "0" : value.isTrue() ? "1" : value.get_str());
+        result.push_back(value.isBool() ? (value.get_bool() ? "1" : "0") : value.get_str());
```

Comment:
not for this PR, but would be good to place the "1" and "0" strings in constants, so we don't hardcode them when they are needed (here, in `SettingToString`, `SettingToInt`, and inside the univalue implementation).

Reply:
I presume `SettingsValue` is supposed to be independent of `UniValue` implementation details. Otherwise we could just return the UniValue internal string representation?

Reply:
ok yeah, agree.
leaving out the `UniValue` implementation, at least all settings methods could share the same constant (`SettingToString`, `ArgsManager::SoftSetBoolArg`, `ArgsManager::GetArgs`).

but.. just a small thing anyway.

Reply:
Maybe a `inline std::string BoolToSetting(bool val) { return val ? "1" : "0"; }` helper?

Reply:
sounds good 👍🏼